### ISSUE TITLE
422 Client Error: Unprocessable Entity for url. (Failing to append domain name)

### ISSUE THREAD

**Issue by nesanton** [theforeman/foreman-ansible-modules#351]
##### SUMMARY

foreman_host module fails to append domain name. In the following line it constructs the fqdn with `domain.fullname`. The Fullname field is a description and could be empty. Should probably be `name` instead.
https://github.com/theforeman/foreman-ansible-modules/blob/3863581bb8f33f4b43a70e457616f012a3c45fca/plugins/modules/foreman_host.py#L131
##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION

```
ansible 2.8.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/usr/share/ansible/installed_modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jun 11 2019, 12:19:05) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
```

##### KATELLO/FOREMAN VERSION

```
tfm-rubygem-katello-3.10.0.48-1.el7sat.noarch
foreman-1.20.1.34-1.el7sat.noarch
```

##### NAILGUN VERSION

```
Version: 0.32.0
```

##### STEPS TO REPRODUCE

run the example snippet against hostgroup with empty Fullname for the domain.

```yaml
- name: "Create a host"
  foreman_host:
    username: "admin"
    password: "changeme"
    server_url: "https://foreman.example.com"
    name: "new_host"
    hostgroup: my_hostgroup
    state: present
```

##### EXPECTED RESULTS

Host is successfully created
##### ACTUAL RESULTS

```
"msg": "Error while creating Host: 422 Client Error: Unprocessable Entity for url: https://foreman.example.com/api/v2/hosts"
```

**Comment by nesanton** [comment]
can do a pull request if you like

**Comment by evgeni** [comment]
@Fobhep @mdellweg could you have a look please? I think @nesanton is spot on here.

**Comment by Fobhep** [comment]
@nesanton seems legit - thank for reporting! Sadly I have currently absolutely no time to take care of it. PR would thus be highly appreciated, alternatively @mdellweg @evgeni can somebody of you take care of this?

**Comment by evgeni** [comment]
Fixed in 827eb5a
